manifests/idesdk/pdk.pp

```diff
@@ -0,0 +1,35 @@
+# pdk.pp
+# Install Puppet Development Kit
+# https://puppet.com/download-puppet-development-kit
+#
+
+class software::idesdk::pdk (
+  $ensure  = $software::params::software_ensure,
+  $version = $software::params::pdk_version,
+  $url     = $software::params::pdk_url,
+) inherits software::params {
+
+  validate_string($ensure)
+
+  case $::operatingsystem {
+    'Ubuntu': {
+      package { 'pdk':
+        ensure   => $ensure,
+        source   => '/tmp/pdk.deb',
+        provider => 'dpkg',
+      }
+
+      file { '/tmp/pdk.deb':
```

Comment:
I don't think we need to download the package for linux any longer.
If you are running puppet on a machine there is a good chance you installed it from the official repos, which also have pdk now. I think we can just get away with the package.

Reply:
Maybe yes. For a pure PDK installation it's still more that just `package {'pdk':}`, though, unfortunately. That's still my motivation. You can look at my [Ansible-driven implementation](https://github.com/painless-software/ansible-software/blob/master/roles/developer/tasks/automation.yml#L4-L29) for comparison. It's 3 steps that you won't recall from the top of your head. They make sense to automate.

Reply:
Can it be optional to download the package?  I will use this and I can make the assumption that the puppet repo is setup on my machines. Can you use a param and do either?